test/e2e_node/node_shutdown_linux_test.go

```diff
@@ -0,0 +1,205 @@
+			nodeName := getNodeName(f)
+			nodeSelector := fields.Set{
+				"spec.nodeName": nodeName,
+			}.AsSelector().String()
+
+			// Define test pods
+			pods := []*v1.Pod{
+				getGracePeriodOverrideTestPod("period-120", nodeName, 120, false),
+				getGracePeriodOverrideTestPod("period-5", nodeName, 5, false),
+				getGracePeriodOverrideTestPod("period-critical-120", nodeName, 120, true),
+				getGracePeriodOverrideTestPod("period-critical-5", nodeName, 5, true),
+			}
+
+			ginkgo.By("Creating batch pods")
+			f.PodClient().CreateBatch(pods)
+
+			list, err := f.PodClient().List(context.TODO(), metav1.ListOptions{
+				FieldSelector: nodeSelector,
+			})
+			framework.ExpectNoError(err)
+			for _, pod := range list.Items {
+				framework.ExpectEqual(
+					pod.Status.Phase == v1.PodRunning,
+					true,
+					"pod is not ready",
+				)
+			}
+
+			ginkgo.By("Emitting shutdown signal")
+			err = emitSignalPrepareForShutdown(true)
+			framework.ExpectNoError(err)
+
+			// Not critical pod should be shutdown
+			gomega.Eventually(func() error {
+				list, err = f.PodClient().List(context.TODO(), metav1.ListOptions{
+					FieldSelector: nodeSelector,
+				})
+				if err != nil {
+					return err
+				}
+				for _, pod := range list.Items {
+					if kubelettypes.IsCriticalPod(&pod) {
+						if pod.Status.Phase != v1.PodRunning {
+							return fmt.Errorf("critical pod should not be shutdown")
+						}
+					} else {
+						if pod.Status.Phase != v1.PodFailed || pod.Status.Reason != "Shutdown" {
+							return fmt.Errorf("pod should be shutdown")
+						}
+
+					}
+				}
+				return nil
+			}, 5*time.Second, 1*time.Second).Should(gomega.BeNil())
+
+			// All pod should be shutdown
+			gomega.Eventually(func() error {
+				list, err = f.PodClient().List(context.TODO(), metav1.ListOptions{
+					FieldSelector: nodeSelector,
+				})
+				if err != nil {
+					return err
+				}
+				for _, pod := range list.Items {
+					if pod.Status.Phase != v1.PodFailed || pod.Status.Reason != "Shutdown" {
+						return fmt.Errorf("pod should be shutdown")
+					}
+				}
+				return nil
+			}, 10*time.Second, 1*time.Second).Should(gomega.BeNil())
+		})
+
+		ginkgo.It("should be able to handle a cancelled shutdown", func() {
+			ginkgo.By("Emitting Shutdown signal")
+			err := emitSignalPrepareForShutdown(true)
+			framework.ExpectNoError(err)
+			gomega.Eventually(func() error {
+				isReady := getNodeReadyStatus(f)
+				if isReady {
+					return fmt.Errorf("node did not become shutdown as expected")
+				}
+				return nil
+			}, 3*time.Second, 1*time.Second).Should(gomega.BeNil())
+
+			ginkgo.By("Emitting Shutdown false signal; cancelling the shutdown")
+			err = emitSignalPrepareForShutdown(false)
+			framework.ExpectNoError(err)
+			gomega.Eventually(func() error {
+				isReady := getNodeReadyStatus(f)
+				if !isReady {
+					return fmt.Errorf("node did not recover as expected")
+				}
+				return nil
+			}, 10*time.Second, 1*time.Second).Should(gomega.BeNil())
```

Comment:
nit regarding poll intervals and timeouts for the `gomega.Eventually`:

Currently, it's not super clear why some of the timeouts are chosen and if their value is significant.  

I would have two suggestions:

1.  If timeout can be anything here, I would suggest bumping this up, say 1 minute since there may be some delay in test say between kubelet and api server and we want to avoid flakes.
2. For the various `gomega.Eventually` poll intervals and timeouts, I would suggest if they can be standardized, and to use the same interval constants throughout the test. E.g. defining some constants and use them throughout. Something like:

```
const(
   nodeStatusPollInterval = 1 * time.Second
   nodeStatusUpdateTimeout = 10 * time.Second
)
```

Reply:
Agree, updated.